daml-assistant/integration-tests/src/DA/Daml/Assistant/IntegrationTests.hs

```diff
@@ -264,6 +265,111 @@ packagingTests = testGroup "packaging"
+          , "init-script: Main:init"
+          , "sandbox-options:"
+          , "  - --wall-clock-time"
+          ]
+        writeFileUTF8 (projDir </> "daml/Main.daml") $ unlines
+          [ "module Main where"
+          , "import Daml.Script"
+          , "template T with p : Party where signatory p"
+          , "init : Script ()"
+          , "init = do"
+          , "  alice <- allocatePartyWithHint \"Alice\" (PartyIdHint \"Alice\")"
+          , "  alice `submit` createCmd (T alice)"
+          , "  pure ()"
+          ]
+        sandboxPort :: Int <- fromIntegral <$> getFreePort
+        jsonApiPort :: Int <- fromIntegral <$> getFreePort
+        let startProc = (shell $ unwords
+              [ "daml"
+              , "start"
+              , "--start-navigator"
+              , "no"
+              , "--sandbox-port"
+              , show sandboxPort
+              , "--json-api-port"
+              , show jsonApiPort
+              ]) {std_in = CreatePipe, cwd = Just projDir}
+        withCurrentDirectory projDir $
+          withCreateProcess startProc $ \startStdIn _ _ startPh ->
+            race_ (waitForProcess' startProc startPh) $
+             do
+              -- The hard-coded secret for testing is "secret".
+              let token =
+                    JWT.encodeSigned
+                      (JWT.HMACSecret "secret")
+                      mempty
+                      mempty
+                        { JWT.unregisteredClaims =
+                            JWT.ClaimsMap $
+                            Map.fromList
+                              [ ( "https://daml.com/ledger-api"
+                                , Aeson.Object $
+                                  HashMap.fromList
+                                    [ ("actAs", Aeson.toJSON ["Alice" :: T.Text])
+                                    , ("ledgerId", "MyLedger")
+                                    , ("applicationId", "foobar")
+                                    ])
+                              ]
+                        }
+              let headers = [("Authorization", "Bearer " <> T.encodeUtf8 token)] :: RequestHeaders
+              waitForHttpServer
+                (threadDelay 100000)
+                ("http://localhost:" <> show jsonApiPort <> "/v1/query")
+                headers
+              writeFileUTF8 (projDir </> "daml/Main.daml") $
+                unlines
+                  [ "module Main where"
+                  , "import Daml.Script"
+                  , "template S with newFieldName : Party where signatory newFieldName"
+                  , "init : Script ()"
+                  , "init = do"
+                  , "  [aliceDetails] <- listKnownParties"
+                  , "  let alice = party aliceDetails"
+                  , "  alice `submit` createCmd (S alice)"
+                  , "  pure ()"
+                  ]
+              maybe (fail "No start process stdin handle") (\h -> hPutChar h 'r' >> hFlush h) startStdIn
```

Comment:
Will this work on Windows? Are do we need to send a separate ENTER press?

Reply:
It should work because of the extra `flush`. The extra `ENTER` just forces the flush on Windows, but we're not checking for an extra `ENTER` to start the rebuild on Windows.

Reply:
Ah, that makes sense. Thanks.